src/chainparamsbase.cpp

```diff
@@ -73,13 +36,13 @@ const CBaseChainParams& BaseParams()
 std::unique_ptr<CBaseChainParams> CreateBaseChainParams(const std::string& chain)
 {
     if (chain == CBaseChainParams::MAIN)
-        return std::unique_ptr<CBaseChainParams>(new CBaseMainParams());
+        return std::unique_ptr<CBaseChainParams>(new CBaseChainParams("", 8332));
     else if (chain == CBaseChainParams::TESTNET)
-        return std::unique_ptr<CBaseChainParams>(new CBaseTestNetParams());
+        return std::unique_ptr<CBaseChainParams>(new CBaseChainParams("testnet3", 18332));
     else if (chain == CBaseChainParams::REGTEST)
-        return std::unique_ptr<CBaseChainParams>(new CBaseRegTestParams());
-    else
-        throw std::runtime_error(strprintf("%s: Unknown chain %s.", __func__, chain));
+        return std::unique_ptr<CBaseChainParams>(new CBaseChainParams("regtest", 18443));
+
+    return std::unique_ptr<CBaseChainParams>(new CBaseChainParams(chain, 18553));
```

Comment:
Isn't the convention for the RPC port to be `P2P port - 1`?

Reply:
You mean ndefault port could be, by default 18554 instead of 18444 ?
Yeah, That would make it a little bit more different than regtest by default but I guess it's alright.
If @MarcoFalke hasn't asked I would have been personally fine with leaving both the same as regtest. These are default ports that can be configured manually anyway.